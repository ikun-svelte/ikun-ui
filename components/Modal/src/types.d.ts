/// <reference types="svelte" />
import type { ClassValue } from 'clsx';

export type KModalProps = {
	show: boolean;
	footer: boolean;
	layout: 'center' | 'right';
	title: string;
	cls: ClassValue;
	attrs: Record<string, string>;
};
