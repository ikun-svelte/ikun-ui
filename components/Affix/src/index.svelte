<script lang="ts">
	import { getPrefixCls } from '@ikun-ui/utils';
	import { clsx } from 'clsx';
	import type { KAffixProps } from './types';
	import type { ScrollTarget } from './utils';
	import { unwrapElement, getScrollTop, getRect, beforeNextFrameOnce } from './utils';
	import { onDestroy, onMount } from 'svelte';
	export let cls: KAffixProps['cls'] = undefined;
	export let attrs: KAffixProps['attrs'] = {};
	export let listenTo: KAffixProps['listenTo'] = undefined;
	export let top: KAffixProps['top'] = undefined;
	export let bottom: KAffixProps['bottom'] = undefined;
	export let triggerTop: KAffixProps['triggerTop'] = undefined;
	export let triggerBottom: KAffixProps['triggerBottom'] = undefined;
	export let positionOption: KAffixProps['positionOption'] = 'fixed';

	let scrollTarget: ScrollTarget | null = null;
	let stickToTopRef = false;
	let stickToBottomRef = false;
	let bottomAffixedTriggerScrollTopRef: number | null = null;
	let topAffixedTriggerScrollTopRef: number | null = null;
	$: affixedRef = stickToBottomRef || stickToTopRef;
	$: mergedOffsetTopRef = triggerTop ?? top;
	$: mergedTopRef = top ?? triggerTop;
	$: mergedBottomRef = bottom ?? triggerBottom;
	$: mergedOffsetBottomRef = triggerBottom ?? bottom;
	let selfRef: Element | null = null;

	const init = (): void => {
		if (listenTo) {
			scrollTarget = unwrapElement(listenTo);
		} else {
			scrollTarget = document;
		}
		if (scrollTarget) {
			scrollTarget.addEventListener('scroll', handleScroll);
			handleScroll();
		}
	};
	function handleScroll(): void {
		beforeNextFrameOnce(doHandleScroll);
	}

	function doHandleScroll(): void {
		const selfEl = selfRef;
		if (!scrollTarget || !selfEl) return;
		const scrollTop = getScrollTop(scrollTarget);
		if (affixedRef) {
			if (topAffixedTriggerScrollTopRef !== null && scrollTop < topAffixedTriggerScrollTopRef) {
				stickToTopRef = false;
				topAffixedTriggerScrollTopRef = null;
			}
			if (
				bottomAffixedTriggerScrollTopRef !== null &&
				scrollTop > bottomAffixedTriggerScrollTopRef
			) {
				stickToBottomRef = false;
				bottomAffixedTriggerScrollTopRef = null;
			}
			return;
		}
		const containerRect = getRect(scrollTarget);
		const affixRect = selfEl.getBoundingClientRect();
		const pxToTop = affixRect.top - containerRect.top;
		const pxToBottom = containerRect.bottom - affixRect.bottom;
		const mergedOffsetTop = mergedOffsetTopRef;
		const mergedOffsetBottom = mergedOffsetBottomRef;
		if (mergedOffsetTop !== undefined && pxToTop <= mergedOffsetTop) {
			stickToTopRef = true;
			topAffixedTriggerScrollTopRef = scrollTop - (mergedOffsetTop - pxToTop);
		} else {
			stickToTopRef = false;
			topAffixedTriggerScrollTopRef = null;
		}
		if (mergedOffsetBottom !== undefined && pxToBottom <= mergedOffsetBottom) {
			stickToBottomRef = true;
			bottomAffixedTriggerScrollTopRef = scrollTop + mergedOffsetBottom - pxToBottom;
		} else {
			stickToBottomRef = false;
			bottomAffixedTriggerScrollTopRef = null;
		}
	}
	onMount(init);
	onDestroy(() => {
		if (!scrollTarget) return;
		scrollTarget.removeEventListener('scroll', handleScroll);
	});

	let styleTop: string = '';
	$: {
		if (stickToTopRef && mergedOffsetTopRef !== undefined && mergedTopRef !== undefined) {
			styleTop = `${mergedTopRef}px`;
		}
	}

	let styleBottom: string = '';
	$: {
		if (stickToBottomRef && mergedOffsetBottomRef !== undefined && mergedBottomRef !== undefined) {
			styleBottom = `${mergedBottomRef}px`;
		}
	}

	const prefixCls = getPrefixCls('affix');
	$: cnames = clsx(
		prefixCls,
		{
			[`${prefixCls}--affixed`]: affixedRef,
			[`${prefixCls}--absolute-positioned`]: positionOption === 'absolute'
		},
		cls
	);
</script>

<div
	class={cnames}
	style:top={styleTop}
	style:bottom={styleBottom}
	bind:this={selfRef}
	{...$$restProps}
	{...attrs}
>
	<slot />
</div>
