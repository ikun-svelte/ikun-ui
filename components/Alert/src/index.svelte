<script lang="ts">
	import type { KAlertProps } from './types';
	import { createEventDispatcher } from 'svelte';
	import { slide } from 'svelte/transition';
	import { quintOut } from 'svelte/easing';
	import { KIcon } from '@ikun-ui/icon';
	import { getPrefixCls } from '@ikun-ui/utils';
	import clsx from 'clsx';

	export let type: KAlertProps['type'] = 'info';
	export let title: KAlertProps['title'] = '';
	export let description: KAlertProps['description'] = '';
	export let showIcon: KAlertProps['showIcon'] = false;
	// will cover type icon
	export let icon: KAlertProps['icon'] = '';
	export let closable: KAlertProps['closable'] = true;
	export let closeIcon: KAlertProps['closeIcon'] = 'i-carbon-close';
	export let cls: KAlertProps['cls'] = undefined;
	export let attrs: KAlertProps['attrs'] = {};

	const dispatch = createEventDispatcher();

	$: visible = true;
	const onClose = (event: Event) => {
		event.stopPropagation();
		visible = false;
		dispatch('close', event);
		handleAnimation();
	};

	// class names
	const prefixCls = getPrefixCls('alert');
	$: baseCls = clsx(prefixCls, `${prefixCls}--${type}`, cls);
	$: iconCls = clsx(`${prefixCls}--icon`);
	$: contentCls = clsx(`${prefixCls}--content`);
	$: titleCls = clsx(`${prefixCls}--title`);
	$: noTitle = !(title || $$slots.title);
	$: descriptionCls = clsx(`${prefixCls}--description`, {
		[`${prefixCls}--description__no-title`]: noTitle
	});
	$: closeCls = clsx(
		`${prefixCls}--close`,
		`${prefixCls}--close__icon-color`,
		{
			[`${prefixCls}--close__no-title`]: noTitle,
			[`${prefixCls}--${type}__close-icon`]: !$$slots.close
		},
		{
			[`${prefixCls}--close__has-description`]: !noTitle && (description || $$slots.default)
		},
		animationCls
	);

	// default type icon
	$: typeIcon = `${prefixCls}--${type}__icon`;
	$: typeIconColor = `${prefixCls}--${type}__icon-color`;
	$: typeIconSize = noTitle ? '18px' : '24px';

	// animation
	let closeRef: HTMLElement | null = null;
	let animationCls = '';
	const handleAnimation = () => {
		if (closeRef) {
			animationCls = `${prefixCls}--${type}__animate`;
			setTimeout(() => {
				animationCls = '';
			}, 310);
		}
	};
</script>

{#if visible}
	<div
		class={baseCls}
		{...$$restProps}
		{...attrs}
		transition:slide={{ delay: 50, easing: quintOut }}
	>
		{#if showIcon && (icon || type || $$slots.icon)}
			<div class={iconCls}>
				<slot name="icon">
					<KIcon
						icon={icon || typeIcon}
						color={typeIconColor}
						width={typeIconSize}
						height={typeIconSize}
					></KIcon>
				</slot>
			</div>
		{/if}
		<div class={contentCls}>
			{#if title || $$slots.title}
				<span class={titleCls}>
					<slot name="title">{title}</slot>
				</span>
			{/if}
			{#if description || $$slots.default}
				<div class={descriptionCls}>
					<slot>{description}</slot>
				</div>
			{/if}
		</div>
		{#if closable}
			<div bind:this={closeRef} class={closeCls} aria-hidden="true" on:click={onClose}>
				<slot name="close">
					<KIcon cls="inline-flex" icon={closeIcon} width="16px" height="16px"></KIcon>
				</slot>
			</div>
		{/if}
	</div>
{/if}
