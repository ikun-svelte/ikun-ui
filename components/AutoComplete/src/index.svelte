<script lang="ts">
	import type { KAutoCompleteProps } from './types';
	import { createEventDispatcher, tick, onMount } from 'svelte';
	import { KInput } from '@ikun-ui/input';
	import { getPrefixCls } from '@ikun-ui/utils';
	import clsx from 'clsx';
	import { isObject } from 'baiwusanyu-utils';
	import { KPopover } from '@ikun-ui/popover';
	import { KVirtualList } from '@ikun-ui/virtual-list';
	import KOption from './option.svelte';

	export let size: KAutoCompleteProps['size'] = 'md';
	export let value: KAutoCompleteProps['value'] = '';
	export let placeholder: KAutoCompleteProps['placeholder'] = '';
	export let disabled: KAutoCompleteProps['disabled'] = false;
	export let append: KAutoCompleteProps['append'] = '';
	export let prepend: KAutoCompleteProps['prepend'] = '';
	export let cls: KAutoCompleteProps['cls'] = undefined;
	export let clsInput: KAutoCompleteProps['cls'] = undefined;
	export let attrs: KAutoCompleteProps['attrs'] = {};
	export let triggerOnFocus: KAutoCompleteProps['triggerOnFocus'] = false;
	export let fetchSuggestions: KAutoCompleteProps['fetchSuggestions'] = undefined;
	export let useCompositionInput: KAutoCompleteProps['useCompositionInput'] = false;

	export let maxHeight: KAutoCompleteProps['maxHeight'] = 250;
	export let fitInputWidth: KAutoCompleteProps['fitInputWidth'] = true;
	export let clearable: KAutoCompleteProps['clearable'] = false;
	// class names
	const prefixCls = getPrefixCls('auto-complete');
	const cname = clsx(prefixCls);
	const dispatch = createEventDispatcher();
	const handleEnter = (e: CustomEvent) => {
		dispatch('enter', e.detail);
	};
	const handleKeydown = (e: CustomEvent) => {
		dispatch('keydown', e.detail);
	};
	const handleChange = (e: CustomEvent) => {
		dispatch('change', e.detail);
	};
	const handleCompositionstart = (e: CustomEvent) => {
		dispatch('compositionstart', e.detail);
	};

	const handleCompositionend = (e: CustomEvent) => {
		dispatch('compositionend', e.detail);
	};

	const handleCompositionInnput = (e: CustomEvent) => {
		dispatch('compositionInput', e.detail);
	};

	const handleTriggerPrepend = (e: CustomEvent) => {
		dispatch('triggerPrepend', e.detail);
	};

	const handleTriggerAppend = (e: CustomEvent) => {
		dispatch('triggerAppend', e.detail);
	};

	// set popover's trigger and content dom width
	let inputSelectRef: HTMLElement | null = null;
	let popoverWidth: undefined | string = undefined;
	let triggerWidth: undefined | string = 'initial';
	let list = [] as Array<Record<string, any>>;
	let text = 'no data';
	const handleInput = async (e: CustomEvent) => {
		curValue = e.detail;
		// 根据输入值重新触发 list 更新
		if (isOpen && fetchSuggestions) {
			curValue = e.detail;
			text = 'loading';
			fetchSuggestions(curValue, (res) => {
				list = res.map((v) => wrapperData(v.value as KAutoCompleteProps['value']));
				setVList();
			});
		}

		if (!triggerOnFocus) {
			isDisabledPopover = false;
			await tick();
			popoverRef.updateShow(true);
		}
	};
	let popoverRef: any = null;
	const setPopoverW = () => {
		if (inputSelectRef) {
			const { width: inputSelectRefWidth } = inputSelectRef.getBoundingClientRect();
			triggerWidth = `${inputSelectRefWidth}px`;
			popoverWidth = fitInputWidth ? `${inputSelectRefWidth}px` : undefined;
		}
	};
	onMount(() => {
		setPopoverW();
	});

	let isDisabledPopover = disabled || !triggerOnFocus;
	$: {
		isDisabledPopover = disabled || !triggerOnFocus;
	}

	let popoverModalRef: HTMLElement | null = null;
	let heightInner = 'initial';
	async function setVList() {
		if (!list.length) {
			text = 'no data';
			return;
		}
		await tick();
		if (popoverModalRef) {
			const container = popoverModalRef.childNodes[0];
			if (container) {
				const { height } = (container as HTMLElement).children[0].getBoundingClientRect();
				if (height > maxHeight) {
					heightInner = `${maxHeight}px`;
				}
			}
		}
	}

	const getLabel = (item: Record<any, string>) => {
		if (isObject(item)) {
			const label = item.label || '';
			return label.toString();
		}
	};

	const wrapperData = (v: KAutoCompleteProps['value']) => {
		return {
			value: v,
			label: v,
			id: v
		};
	};

	$: curValue = value;
	let isOpen = false;
	async function onOpen(e: CustomEvent) {
		isOpen = e.detail;
		if (fetchSuggestions && (triggerOnFocus || (!triggerOnFocus && !isDisabledPopover))) {
			fetchSuggestions(curValue, (res) => {
				list = res.map((v) => wrapperData(v.value as KAutoCompleteProps['value']));
				setVList();
			});
		}
	}

	let inputRef = null;
	// updateValue
	const handleSelect = async (data: Record<string, any>) => {
		dispatch('updateValue', data.value);
		curValue = data.value;
		if (inputRef) {
			inputRef.doUpdateFormField(curValue);
		}
		popoverRef.updateShow(false);
	};

	const noDataCls = clsx(`k-select--tx__empty`);
</script>

<KPopover
	trigger="click"
	bind:this={popoverRef}
	disabled={isDisabledPopover}
	clsTrigger={cls}
	cls="px-0"
	arrow={false}
	on:change={onOpen}
	width={triggerWidth}
	placement="bottom"
>
	<div class={cname} {...attrs} slot="triggerEl" bind:this={inputSelectRef} aria-hidden="true">
		<KInput
			bind:this={inputRef}
			on:input={handleInput}
			on:enter={handleEnter}
			on:keydown={handleKeydown}
			on:change={handleChange}
			on:compositionstart={handleCompositionstart}
			on:compositionend={handleCompositionend}
			on:compositionInput={handleCompositionInnput}
			on:triggerPrepend={handleTriggerPrepend}
			on:triggerAppend={handleTriggerAppend}
			{size}
			value={curValue}
			{placeholder}
			{disabled}
			{append}
			{prepend}
			{clearable}
			cls={clsInput}
			{useCompositionInput}
			{...$$restProps}
			{...attrs}
			type="text"
		>
			<slot name="prefix" slot="prefix" />
			<slot name="suffix" slot="suffix" />
		</KInput>
	</div>
	<div
		slot="contentEl"
		bind:this={popoverModalRef}
		style:overflow-y="auto"
		style:width={popoverWidth}
		style:min-width={triggerWidth}
		style:height={heightInner}
		style:max-height={`${maxHeight}px`}
	>
		{#if list.length > 0}
			<KVirtualList data={list} key="id" estimateSize={list.length} let:data>
				{#if !$$slots.default}
					<KOption {fitInputWidth} label={getLabel(data)} on:click={() => handleSelect(data)}
					></KOption>
				{:else}
					<slot {data} onSelect={handleSelect} label={getLabel(data)} />
				{/if}
			</KVirtualList>
		{:else}
			<p class={noDataCls}>{text}</p>
		{/if}
	</div>
</KPopover>
