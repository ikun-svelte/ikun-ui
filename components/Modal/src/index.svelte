<script lang="ts">
	import { KButton } from '@ikun-ui/button';
	import { KClientOnly } from '@ikun-ui/client-only';
	import { KMask } from '@ikun-ui/mask';
	import { KIcon } from '@ikun-ui/icon';
	import { createEventDispatcher } from 'svelte';
	import { fly } from 'svelte/transition';
	import { clsx } from 'clsx';
	import type { KModalProps } from './types';

	export let show: KModalProps['show'] = false;
	export let attrs: KModalProps['attrs'] = {};
	export let cls: KModalProps['cls'] = '';
	export let title: KModalProps['title'] = '';
	export let footer: KModalProps['footer'] = false;
	// center、right
	export let layout: KModalProps['layout'] = 'center';
	// close、cancel、confirm
	const dispatch = createEventDispatcher();
	$: showInner = show;
	const close = () => {
		showInner = !showInner;
		dispatch('close', showInner);
	};
	const cancel = () => {
		dispatch('cancel');
		close();
	};

	const confirm = () => {
		dispatch('confirm');
		close();
	};

	$: cnames = clsx('k-modal--base k-modal--base__dark', cls);
	$: footerCls = clsx('k-modal--footer', layout === 'center' ? 'justify-center' : 'justify-end');
</script>

<KClientOnly>
	<KMask value={showInner} target={document.body}>
		<div
			{...attrs}
			class={cnames}
			out:fly={{ y: -30, duration: 300 }}
			in:fly={{ y: -30, duration: 300 }}
		>
			<slot name="header">
				<div class="k-modal--header">
					<span class="k-modal--header--title k-modal--header--title__dark">
						{title}
					</span>
					<KIcon icon="i-carbon-close" on:click={close} btn color="k-modal--header--icon__hover" />
				</div>
			</slot>
			<div class="k-modal--body">
				<slot />
			</div>
			{#if footer}
				<slot name="footer">
					<div class={footerCls}>
						<KButton cls="k-modal--footer--btn" on:click={cancel} type="info">Cancel</KButton>
						<KButton cls="k-modal--footer--btn" on:click={confirm} type="primary">Confirm</KButton>
					</div>
				</slot>
			{/if}
		</div>
	</KMask>
</KClientOnly>
