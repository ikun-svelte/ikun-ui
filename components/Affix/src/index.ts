/// <reference types="./types" />
import Affix from './index.svelte';
export { Affix as KAffix };

export default Affix;
