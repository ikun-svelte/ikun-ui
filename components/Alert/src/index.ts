/// <reference types="./types" />
import Alert from './index.svelte';
export { Alert as KAlert };

export default Alert;

export type { KAlertProps } from './types';
