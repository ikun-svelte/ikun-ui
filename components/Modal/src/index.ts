/// <reference types="./types" />
import Modal from './index.svelte';

export default Modal;

export { Modal as KModal };

export type { KModalProps } from './types';
