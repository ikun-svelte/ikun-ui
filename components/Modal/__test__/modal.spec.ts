import { tick } from 'svelte';
import { afterEach, expect, test, vi, describe, beforeEach } from 'vitest';
import { KModal } from '../src/index';
import KModalSlotfrom from './fixture/modal.slot.svelte';

let host: HTMLElement;
const initHost = () => {
	host = document.createElement('div');
	host.setAttribute('id', 'host');
	document.body.appendChild(host);
};
beforeEach(() => {
	initHost();
	vi.useFakeTimers();
});
afterEach(() => {
	host.remove();
	vi.restoreAllMocks();
});

describe('Test: KModal', () => {
	test('options: show', async () => {
		const instance = new KModal({
			target: host,
			props: {
				title: '🐔 ！！',
				show: false
			}
		});
		await tick();
		expect(instance).toBeTruthy();
		expect(host.innerHTML.includes('k-modal--base')).not.toBeTruthy();
		instance.$set({ show: true });
		await tick();
		await vi.advanceTimersByTimeAsync(300);
		expect(host.innerHTML.includes('k-modal--base')).toBeTruthy();
	});

	test('options: title', async () => {
		const instance = new KModal({
			target: host,
			props: {
				title: '🐔',
				show: true
			}
		});
		await tick();
		expect(instance).toBeTruthy();
		expect(host.innerHTML.includes('🐔')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('options: footer', async () => {
		const instance = new KModal({
			target: host,
			props: {
				title: '🐔',
				show: true,
				footer: true
			}
		});
		await tick();
		expect(instance).toBeTruthy();
		expect(host.innerHTML.includes('k-modal--footer')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('options: layout & center', async () => {
		const instance = new KModal({
			target: host,
			props: {
				title: '🐔',
				show: true,
				footer: true,
				layout: 'center'
			}
		});
		await tick();
		expect(instance).toBeTruthy();
		expect(host.innerHTML.includes('k-modal--footer justify-center')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('options: layout & right', async () => {
		const instance = new KModal({
			target: host,
			props: {
				title: '🐔',
				show: true,
				footer: true,
				layout: 'right'
			}
		});
		await tick();
		expect(instance).toBeTruthy();
		expect(host.innerHTML.includes('k-modal--footer justify-end')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});

	test('event: should trigger cancel event', async () => {
		const mockFn = vi.fn();
		const instance = new KModal({
			target: host,
			props: {
				footer: true,
				show: true
			}
		});
		await tick();
		instance.$on('cancel', mockFn);
		const btn = host.querySelector('.k-button--info__fill');
		btn!.dispatchEvent(new Event('click', { bubbles: true }));
		await tick();
		expect(instance).toBeTruthy();
		expect(mockFn).toBeCalled();
	});

	test('event: should trigger confirm  event', async () => {
		const mockFn = vi.fn();
		const instance = new KModal({
			target: host,
			props: {
				footer: true,
				show: true
			}
		});
		await tick();
		instance.$on('confirm', mockFn);
		const btn = host.querySelector('.k-button--primary__fill');
		btn!.dispatchEvent(new Event('click', { bubbles: true }));
		await tick();
		expect(instance).toBeTruthy();
		expect(mockFn).toBeCalled();
	});

	test('event: should trigger close event', async () => {
		const mockFn = vi.fn();
		const instance = new KModal({
			target: host,
			props: {
				footer: true,
				show: true
			}
		});
		await tick();
		instance.$on('close', mockFn);
		const btn = host.querySelector('.i-carbon-close');
		btn!.dispatchEvent(new Event('click', { bubbles: true }));
		await tick();
		expect(instance).toBeTruthy();
		expect(mockFn).toBeCalled();
	});

	test('slot: custom render', async () => {
		const instance = new KModalSlotfrom({
			target: host
		});
		await tick();
		expect(instance).toBeTruthy();
		expect(host.innerHTML.includes('🐔')).toBeTruthy();
		expect(host.innerHTML.includes('💕')).toBeTruthy();
		expect(host.innerHTML.includes('🏀')).toBeTruthy();
		expect(host.innerHTML).matchSnapshot();
	});
});
