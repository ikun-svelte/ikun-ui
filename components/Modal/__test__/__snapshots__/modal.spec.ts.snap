// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test: KModal > options: footer 1`] = `"<div class=\\"k-mask--base\\" style=\\"top: 0px; left: 0px; width: 100%; height: 100%; position: fixed;\\"><div class=\\"k-modal--base k-modal--base__dark\\"><div class=\\"k-modal--header\\"><span class=\\"k-modal--header--title k-modal--header--title__dark\\">🐔</span> <span class=\\"k-icon--base k-icon--role-button k-icon-transition i-carbon-close k-modal--header--icon__hover\\" role=\\"button\\" aria-hidden=\\"true\\" style=\\"width: 24px; height: 24px;\\"></span></div> <div class=\\"k-modal--body\\"></div> <div class=\\"k-modal--footer justify-center\\"><button style=\\"border-radius: 4px; font-size: 16px;\\" class=\\"k-button k-button--base k-button--info__fill k-button--info__hover__fill k-button--info__active k-button--info__focus k-modal--footer--btn\\" aria-hidden=\\"true\\">  Cancel </button> <button style=\\"border-radius: 4px; font-size: 16px;\\" class=\\"k-button k-button--base k-button--primary__fill k-button--primary__hover__fill k-button--primary__active k-button--primary__focus k-modal--footer--btn\\" aria-hidden=\\"true\\">  Confirm </button></div></div></div>"`;

exports[`Test: KModal > options: layout & center 1`] = `"<div class=\\"k-mask--base\\" style=\\"top: 0px; left: 0px; width: 100%; height: 100%; position: fixed;\\"><div class=\\"k-modal--base k-modal--base__dark\\"><div class=\\"k-modal--header\\"><span class=\\"k-modal--header--title k-modal--header--title__dark\\">🐔</span> <span class=\\"k-icon--base k-icon--role-button k-icon-transition i-carbon-close k-modal--header--icon__hover\\" role=\\"button\\" aria-hidden=\\"true\\" style=\\"width: 24px; height: 24px;\\"></span></div> <div class=\\"k-modal--body\\"></div> <div class=\\"k-modal--footer justify-center\\"><button style=\\"border-radius: 4px; font-size: 16px;\\" class=\\"k-button k-button--base k-button--info__fill k-button--info__hover__fill k-button--info__active k-button--info__focus k-modal--footer--btn\\" aria-hidden=\\"true\\">  Cancel </button> <button style=\\"border-radius: 4px; font-size: 16px;\\" class=\\"k-button k-button--base k-button--primary__fill k-button--primary__hover__fill k-button--primary__active k-button--primary__focus k-modal--footer--btn\\" aria-hidden=\\"true\\">  Confirm </button></div></div></div>"`;

exports[`Test: KModal > options: layout & right 1`] = `"<div class=\\"k-mask--base\\" style=\\"top: 0px; left: 0px; width: 100%; height: 100%; position: fixed;\\"><div class=\\"k-modal--base k-modal--base__dark\\"><div class=\\"k-modal--header\\"><span class=\\"k-modal--header--title k-modal--header--title__dark\\">🐔</span> <span class=\\"k-icon--base k-icon--role-button k-icon-transition i-carbon-close k-modal--header--icon__hover\\" role=\\"button\\" aria-hidden=\\"true\\" style=\\"width: 24px; height: 24px;\\"></span></div> <div class=\\"k-modal--body\\"></div> <div class=\\"k-modal--footer justify-end\\"><button style=\\"border-radius: 4px; font-size: 16px;\\" class=\\"k-button k-button--base k-button--info__fill k-button--info__hover__fill k-button--info__active k-button--info__focus k-modal--footer--btn\\" aria-hidden=\\"true\\">  Cancel </button> <button style=\\"border-radius: 4px; font-size: 16px;\\" class=\\"k-button k-button--base k-button--primary__fill k-button--primary__hover__fill k-button--primary__active k-button--primary__focus k-modal--footer--btn\\" aria-hidden=\\"true\\">  Confirm </button></div></div></div>"`;

exports[`Test: KModal > options: title 1`] = `"<div class=\\"k-mask--base\\" style=\\"top: 0px; left: 0px; width: 100%; height: 100%; position: fixed;\\"><div class=\\"k-modal--base k-modal--base__dark\\"><div class=\\"k-modal--header\\"><span class=\\"k-modal--header--title k-modal--header--title__dark\\">🐔</span> <span class=\\"k-icon--base k-icon--role-button k-icon-transition i-carbon-close k-modal--header--icon__hover\\" role=\\"button\\" aria-hidden=\\"true\\" style=\\"width: 24px; height: 24px;\\"></span></div> <div class=\\"k-modal--body\\"></div> </div></div>"`;

exports[`Test: KModal > slot: custom render 1`] = `"<div class=\\"k-mask--base\\" style=\\"top: 0px; left: 0px; width: 100%; height: 100%; position: fixed;\\"><div class=\\"k-modal--base k-modal--base__dark\\"><span class=\\"flex\\" slot=\\"header\\">🐔</span> <div class=\\"k-modal--body\\"><p class=\\"text-center !my-2\\">💕</p></div> <div class=\\"fc w-full\\" slot=\\"footer\\">🏀</div></div></div>"`;
